Blank operations - Trim Trailing

Hikmetcan-dev opened the pull request.

Hikmetcan-dev committed:
Trim Trailing Operation

Hikmetcan-dev committed:
Updated Trim Trailing

Jazzco commented on src/editors/codeedit.cpp:
```diff
@@ -2853,5 +2853,29 @@ void CodeEdit::moveLines(bool moveLinesUp)
+void CodeEdit::trimtrailing()
+{
+    QTextCursor c= textCursor();
+    c.movePosition(c.Start);
```
Instead of "c.Start" this should be "QTextCursor::Start"  ("c" is an instance, but statics like this enum should be referenced with the class itself)

Jazzco commented on src/editors/codeedit.cpp:
```diff
@@ -2853,5 +2853,29 @@ void CodeEdit::moveLines(bool moveLinesUp)
+    QTextCursor c= textCursor();
+    c.movePosition(c.Start);
+    c.movePosition(c.EndOfLine);
+    for(int i=0;i<blockCount();i++){
```
blockCount() here iterates over the whole document. You probably wanted to iterate over the selection: c.block() gives the block of the position. For the anchor you can create a second QTextCursor, set it's position to c.anchor() and get the block of it.

Hikmetcan-dev committed:
Trim Trailing w/ selection

Hikmetcan-dev committed:
Trim Trailing Code Revision

Added Trim Helper function to prevent duplicate codes that will be used in Trim Trailing, Trim Leading and Trim Leading and Trailing.
Variable names and codes are revised and comments are added for a better readability.

Hikmetcan-dev committed:
Trim Leading, Trim Leading and Trailing add

Added Trim Leading, Trim Leading and Trailing functions.
Revised code for better readability.

Hikmetcan-dev committed:
EOL to Space

EOL to Space function added.
This function changes the new line operator to space character in a given range or all of the text.

Hikmetcan-dev committed:
Tab to Space

Added Tab to Space function.

For some reason (possibly related to QT) Gams Studio turns tab into 3 consecutive spaces. That is the reason I used "   ".
